src/mbgl/style/layers/background_layer_properties.hpp

```diff
@@ -30,5 +32,20 @@ class BackgroundPaintProperties : public Properties<
     BackgroundOpacity
 > {};
 
+class BackgroundPropertiesWrapper final : public LayerPropertiesWrapper {
```

Comment:
Technically, looks good, except the name :smile: It is not a simple wrapper anymore, as it contains original style layer and few other things.

Reply:
Yeah, I've been looking for a good one and did not find yet.. any options? From OOP point of view it is "Memento", however `LayerPropertiesMemento` does not look good :smile:

Reply:
Let's call it just `LayerProperties` and e.g. `CircleLayerProperties`..